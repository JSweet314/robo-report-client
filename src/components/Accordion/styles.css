.parent-acc {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 2rem auto;
  background-color: white;
  border: 1px solid black;
}

.acc-heading {
  padding: .5em 0;
  font-size: 2rem;
  font-weight: 600;
  width: 90%;
  text-align: center
}

.acc-content {
  display: grid;
  grid-column-gap: 1rem;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: auto;
}

.acc-close {
  display: none;
}

.acc-open {
  width: fit-content;
  font-size: 1.7rem;
  background-color: white;
  padding: 0rem 2rem 2rem;
  border-radius: 15px;
}

.report-heading-group {
  display: flex;
  justify-content: flex-start;
  align-items: center;
}

.report-heading {
  width: 100%;
  align-items: center;
  text-align: left;
}

.accordion-symbol {
  height: 3rem;
  align-self: flex-end;
}
