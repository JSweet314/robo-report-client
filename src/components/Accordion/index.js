import React, { Component } from 'react';
import PropTypes from 'prop-types';
import plusSymbol from '../../assets/images/plus-symbol.svg';
import minusSymbol from '../../assets/images/minus-symbol.svg';
import './styles.css';

class Accordion extends Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
      className: 'acc-content acc-close',
      headingClassName: 'acc-heading'
    };
  }

  handleClick = () => {
    const { open } = this.state;
    if (open) {
      this.setState({
        open: false,
        className: 'acc-content acc-close',
        headingClassName: 'acc-heading'
      });
    } else {
      this.setState({
        open: true,
        className: 'acc-content acc-open',
        headingClassName: 'acc-heading clicked'
      });
    }
  };

  render() {
    const { content, headingText } = this.props;
    const { className, headingClassName, open } = this.state;
    const symbol = open ? minusSymbol : plusSymbol;
    return (
      <div className="parent-acc">
        <div className={headingClassName} onClick={this.handleClick}>
          <div className='report-heading-group'>
            <p className='report-heading'>
              {headingText}
            </p>
            <img className='accordion-symbol' src={symbol} alt="plus/minus"/>
          </div>
        </div>
        <div className={className}>{content}</div>
      </div>
    );
  }
}

Accordion.propTypes = {
  content: PropTypes.array.isRequired,
  headingText: PropTypes.string.isRequired
};

export default Accordion;
