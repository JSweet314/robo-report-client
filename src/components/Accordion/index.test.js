import React from 'react';
import { shallow } from 'enzyme';
import Accordion from './';

describe('Accordion', () => {
  let wrapper;
  const mockContent = [<p key={1}>Content</p>, <p key={2}>Moar Content</p>];
  const mockHeadingText = 'Robo Calls 2018-06-02 555-555-5555';

  beforeEach(() => {
    wrapper = shallow(
      <Accordion content={mockContent} headingText={mockHeadingText} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot', () => {
    wrapper.instance().handleClick();

    expect(wrapper).toMatchSnapshot();
  });
});
