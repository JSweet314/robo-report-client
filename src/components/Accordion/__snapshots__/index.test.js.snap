// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should match the snapshot 1`] = `
<div
  className="parent-acc"
>
  <div
    className="acc-heading"
    onClick={[Function]}
  >
    <div
      className="report-heading-group"
    >
      <p
        className="report-heading"
      >
        Robo Calls 2018-06-02 555-555-5555
      </p>
      <img
        alt="plus/minus"
        className="accordion-symbol"
        src="plus-symbol.svg"
      />
    </div>
  </div>
  <div
    className="acc-content acc-close"
  >
    <p
      key="1"
    >
      Content
    </p>
    <p
      key="2"
    >
      Moar Content
    </p>
  </div>
</div>
`;

exports[`Accordion should match the snapshot 2`] = `
<div
  className="parent-acc"
>
  <div
    className="acc-heading"
    onClick={[Function]}
  >
    <div
      className="report-heading-group"
    >
      <p
        className="report-heading"
      >
        Robo Calls 2018-06-02 555-555-5555
      </p>
      <img
        alt="plus/minus"
        className="accordion-symbol"
        src="plus-symbol.svg"
      />
    </div>
  </div>
  <div
    className="acc-content acc-close"
  >
    <p
      key="1"
    >
      Content
    </p>
    <p
      key="2"
    >
      Moar Content
    </p>
  </div>
</div>
`;
